File: src/main/java/com/illuzor/afo/actions/MainModuleEnterAction.java
package com.illuzor.afo.actions;

import com.illuzor.afo.constants.Id;
import com.intellij.ide.util.PropertiesComponent;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;
import org.jetbrains.annotations.NotNull;

import java.io.File;

public class MainModuleEnterAction extends ActionBase {

    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        Project project = e.getData(PlatformDataKeys.PROJECT);
        String moduleName = showInput(project);

        if (moduleName == null || moduleName.isEmpty()) {
            return;
        }

        String path = project.getBasePath() + "/" + moduleName + "/build.gradle";
        String ktsPath = path + ".kts";
        if (!new File(path).exists() && !new File(ktsPath).exists()) {
            showError("Module '" + moduleName + "' Does Not Exists");
            return;
        }
        PropertiesComponent.getInstance(project).setValue(Id.MAIN_MODULE_KEY, moduleName);
    }

    private String showInput(Project project) {
        return Messages.showInputDialog(
                project,
                "Enter Main Module Name",
                "Main Module Name:",
                Messages.getQuestionIcon()
        );
    }
}

File: src/main/kotlin/com/illuzor/afo/actions/define/MainModuleEnterAction.kt
package com.illuzor.afo.actions.define

import com.illuzor.afo.ext.notExists
import com.illuzor.afo.ui.showErrorDialog
import com.intellij.openapi.ui.Messages
import java.io.File

internal class MainModuleEnterAction : DefineMainModuleBaseAction() {

    override fun perform() {
        val moduleName = showInput()?.replace(':', '/')
        if (moduleName.isNullOrEmpty()) {
            return
        }

        val modulePath = "$projectPath/$moduleName"
        val moduleFolder = File(modulePath)
        if (moduleFolder.notExists()) {
            showErrorDialog("Directory '$modulePath' does not exist")
            return
        }

        checkModuleFolderAndSave(moduleFolder, moduleName)
    }

    private fun showInput(): String? =
        Messages.showInputDialog(
            project,
            "Enter main module name. For example 'module' or 'module:submodule'",
            "Main Module Name:",
            Messages.getQuestionIcon(),
        )
}
